### DISCUSSION TITLE
Routerify-lite

### DISCUSSION THREAD

**Discussion by kingluo** [routerify/routerify#81] [Show and tell]
I try to rewrite some code paths to improve the performance (https://github.com/routerify/routerify/issues/77).
Check my repo please:
https://github.com/kingluo/routerify-lite

**Comment by seanpianka** [comment]
Very interesting! How does it fair when running against the Routerify integration tests or examples?

**Reply by kingluo** [reply]
I add the gist link of the codes for the benchmark. In fact, for the routerify one, it originates from the routerify example.

**Reply by gsserge** [reply]
Nice! What code paths did you optimize to get better performance?

**Reply by kingluo** [reply]
Well, because routerify-litet only supports path matching and error handling, so the code is straightforward. For example, it does not support middleware so it does not do `match_regex_set` which creates four vectors.

**Reply by kingluo** [reply]
I use `cargo flamegraph` to check the hot path. And I found the RegexSet is a bit expensive. I have an idea: if all paths do not include regular expressions and `:param`, then we could use a hash table (BTreeMap to reserve order) to match path?

**Reply by seanpianka** [reply]
This can be gated with a disabled-by-default feature flag for simplified matching behavior that interested users can disable for this performance gain. @kingluo, would you be interested in opening a pull request that enables this type of matching behavior with a feature flag?

**Comment by kingluo** [comment]
I add plain mode (optional) in routerify-lite. This mode uses HashMap to do path matching. The improved performance is obvious when the pattern list is big.

https://github.com/kingluo/routerify-lite#performance-2
